<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string> CyberScanner</string>
  </property>
  <property name="windowIcon">
    <iconset resource="resources.qrc">
      <normaloff>:/icons/Images/CyberScanner.png</normaloff>
    </iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_main">
    <item>
     <widget class="QGroupBox" name="groupBox_input">
      <property name="title">
       <string>Scan Configuration</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_input">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_target">
         <item>
          <widget class="QLabel" name="label_target">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>Target Host/IP:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_target">
           <property name="placeholderText">
            <string>192.168.1.1 or example.com</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_presets">
           <property name="minimumSize">
            <size>
             <width>120</width>
             <height>0</height>
            </size>
           </property>
           <item>
            <property name="text">
             <string>Quick Targets</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>localhost</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>192.168.1.1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>8.8.8.8</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_ports">
         <item>
          <widget class="QLabel" name="label_ports">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>Port Range:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_portFrom">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>65535</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_to">
           <property name="text">
            <string>to</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_portTo">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>65535</number>
           </property>
           <property name="value">
            <number>1000</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_or">
           <property name="text">
            <string>or Custom:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_customPorts">
           <property name="placeholderText">
            <string>80,443,8080 or 1-100,443,8000-9000</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_portPresets">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>0</height>
            </size>
           </property>
           <item>
            <property name="text">
             <string>Port Presets</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Web (80,443,8080,8443)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Mail (25,110,143,993,995)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Common (21,22,23,25,53,80,110,443)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>All (1-65535)</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_options">
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_buttons">
         <item>
          <widget class="QPushButton" name="pushButton_start">
           <property name="minimumSize">
            <size>
             <width>120</width>
             <height>35</height>
            </size>
           </property>
           <property name="styleSheet">
            <string>QPushButton {
    background-color: #4CAF50;
    color: white;
    font-weight: bold;
    border: none;
    border-radius: 4px;
}
QPushButton:hover {
    background-color: #66BB6A;
}
QPushButton:pressed {
    background-color: #388E3C;
}</string>
           </property>
           <property name="text">
            <string>START SCAN</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_stop">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimumSize">
            <size>
             <width>80</width>
             <height>35</height>
            </size>
           </property>
           <property name="styleSheet">
            <string>QPushButton {
    background-color: #f44336;
    color: white;
    font-weight: bold;
    border: none;
    border-radius: 4px;
}
QPushButton:hover {
    background-color: #EF5350;
}
QPushButton:pressed {
    background-color: #D32F2F;
}</string>
           </property>
           <property name="text">
            <string>STOP</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_clear">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>35</height>
            </size>
           </property>
           <property name="text">
            <string>CLEAR RESULTS</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_progress">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>100</height>
       </size>
      </property>
      <property name="title">
       <string>Scan Progress</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_progress">
       <item>
        <widget class="QProgressBar" name="progressBar">
         <property name="value">
          <number>0</number>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_status">
         <item>
          <widget class="QLabel" name="label_status">
           <property name="text">
            <string>Status: Ready</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="label_stats">
           <property name="text">
            <string>Scanned: 0 | Open: 0 | Time: 00:00</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_results">
       <attribute name="title">
        <string>Results</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_results">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_filter">
          <item>
           <widget class="QLabel" name="label_filter">
            <property name="text">
             <string>Filter:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEdit_filter">
            <property name="placeholderText">
             <string>Filter results by port, service, or status...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_filterType">
            <item>
             <property name="text">
              <string>All</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Open Only</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Closed Only</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTableWidget" name="tableWidget_results">
          <property name="styleSheet">
           <string>QTableWidget {
    background-color: #ffffff;
    alternate-background-color: #f5f5f5;
    color: #333333;
    gridline-color: #ddd;
    selection-background-color: #0078d4;
    selection-color: white;
}
QTableWidget::item {
    padding: 5px;
    border: none;
}
QTableWidget::item:selected {
    background-color: #0078d4;
    color: white;
}
QTableWidget::item:alternate {
    background-color: #f8f9fa;
}
QHeaderView::section {
    background-color: #e9ecef;
    color: #495057;
    border: 1px solid #dee2e6;
    padding: 8px;
    font-weight: bold;
}
QHeaderView::section:hover {
    background-color: #dee2e6;
}</string>
          </property>
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
          </property>
          <property name="sortingEnabled">
           <bool>true</bool>
          </property>
          <column>
           <property name="text">
            <string>Port</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Protocol</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Status</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Service</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Banner</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Response Time</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_log">
       <attribute name="title">
        <string>Log Output</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_log">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_logControls">
          <item>
           <widget class="QCheckBox" name="checkBox_autoScroll">
            <property name="text">
             <string>Auto-scroll</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_clearLog">
            <property name="text">
             <string>Clear Log</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_saveLog">
            <property name="text">
             <string>Save Log</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTextEdit" name="textEdit_log">
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>9</pointsize>
           </font>
          </property>
          <property name="styleSheet">
           <string>QTextEdit {
    background-color: #2b2b2b;
    color: #ffffff;
    border: 1px solid #555555;
    font-family: 'Consolas', 'Monaco', monospace;
}
QScrollBar:vertical {
    background: #3c3c3c;
    width: 12px;
    border-radius: 6px;
}
QScrollBar::handle:vertical {
    background: #606060;
    border-radius: 6px;
    min-height: 20px;
}
QScrollBar::handle:vertical:hover {
    background: #707070;
}</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>900</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionGithub"/>
   </widget>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionExport_CSV">
   <property name="text">
    <string>Export to CSV</string>
   </property>
  </action>
  <action name="actionExport_XML">
   <property name="text">
    <string>Export to XML</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionGithub">
   <property name="text">
    <string>My Github</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
